c7n/resources/iam.py

```diff
@@ -985,35 +985,104 @@ def process(self, resources, event=None):
+    a given list:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: iam-roles-with-selected-policies
+            resource: aws.iam-role
+            filters:
+              - type: has-specific-managed-policy
+                value:
+                  - AmazonS3FullAccess
+                  - AWSOrganizationsFullAccess
+
+    :example:
+
+    Check for roles with attached policy names matching a pattern:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: iam-roles-with-full-access-policies
+            resource: aws.iam-role
+            filters:
+              - type: has-specific-managed-policy
+                op: glob
+                value: "*FullAccess"
+
+    Check for roles with attached policy ARNs matching a pattern:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: iam-roles-with-aws-full-access-policies
+            resource: aws.iam-role
+            filters:
+              - type: has-specific-managed-policy
+                key: PolicyArn
+                op: regex
+                value: "arn:aws:iam::aws:policy/.*FullAccess"
     """
 
-    schema = type_schema('has-specific-managed-policy', value={'type': 'string'})
+    schema = type_schema('has-specific-managed-policy', rinherit=ValueFilter.schema)
     permissions = ('iam:ListAttachedRolePolicies',)
+    annotation_key = 'c7n:AttachedPolicies'
+    matched_annotation_key = 'c7n:MatchedPolicies'
+    schema_alias = False
 
-    def _managed_policies(self, client, resource):
-        return [r['PolicyName'] for r in client.list_attached_role_policies(
-            RoleName=resource['RoleName'])['AttachedPolicies']]
+    def __init__(self, data, manager=None):
+        # Preserve backward compatibility
+        if 'key' not in data:
+            data['key'] = 'PolicyName'
+        super(SpecificIamRoleManagedPolicy, self).__init__(data, manager)
+
+    def get_managed_policies(self, client, role_set):
+        for role in role_set:
+            role[self.annotation_key] = [
+                role_policy
+                for role_policy
+                in client.list_attached_role_policies(
+                    RoleName=role['RoleName'])['AttachedPolicies']
+            ]
 
     def process(self, resources, event=None):
-        c = local_session(self.manager.session_factory).client('iam')
-        if self.data.get('value'):
-            return [r for r in resources if self.data.get('value') in self._managed_policies(c, r)]
-        return []
+        client = local_session(self.manager.session_factory).client('iam')
+        with self.executor_factory(max_workers=2) as w:
+            augment_set = [r for r in resources if self.annotation_key not in r]
+            self.log.debug(
+                "Querying %d roles' attached policies" % len(augment_set))
+            list(w.map(
+                functools.partial(self.get_managed_policies, client),
+                chunks(augment_set, 50)))
```

Comment:
`chunks()` doesn't seem necessary here because it's not helping us batch calls (which is a common reason to chunk resource sets elsewhere). It does work though. And while it adds a layer of nesting and chance of partial failures to `get_managed_policies()`, as far as I can tell a partial failure there would blow up a policy run regardless of whether it was handling 1 or 50 policies at a time...